common/Resources/ti.internal/extensions/ti/ti.ui.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2019 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+let colorset;
+let osVersion;
+
+// As Android passes a new instance of Ti.UI to every JS file we can't just
+// Ti.UI within this file, we must call kroll.binding to get the Titanium
+// namespace that is passed in with require and that deal with the .UI
+// namespace that is on that directly.
+let uiModule = Ti.UI;
+if (Ti.Android) {
+	uiModule = kroll.binding('Titanium').Titanium.UI;
+}
+
+uiModule.SEMANTIC_COLOR_TYPE_LIGHT = 'light';
+uiModule.SEMANTIC_COLOR_TYPE_DARK = 'dark';
+uiModule.semanticColorType = Ti.UI.SEMANTIC_COLOR_TYPE_LIGHT;
+
+uiModule.fetchSemanticColor = function fetchSemanticColor (colorName) {
+	if (!osVersion) {
+		osVersion = parseInt(Ti.Platform.version.split('.')[0]);
+	}
+
+	if (Ti.App.iOS && osVersion >= 11) {
```

Comment:
If I run on 11 < iOS < 13, color is not getting changed for dark and light mode set from Ti.UI.semanticColorType. Here osVersion check should be 
`if (Ti.App.iOS && osVersion >= 13) { }` 
.

Reply:
So the API we use [colorNamed](https://developer.apple.com/documentation/uikit/uicolor/2877380-colornamed?language=objc) is available from iOS 11 onwards, I figured it was always better to fall back to the native API if possible rather than our mocked dark mode, but yeah it only returns light colors. I'll make this check 13+ so we can handle that

Reply:
Updated